### PR TITLE
Fetchdata

### LINKED ISSUE #30: get data from the database
As a user, I want to see the available properties on the map 
acceptance criteria :
- [ ] xhr request to the database
- [ ] a function that updates state with the data from the database
- [ ] property details on the marker popup 
- [ ] when the user clicks the popup they get more details about the property on a single page

### PR DESCRIPTION
**Pull request by hoslack**
Closes #30

### CODE REVIEW

File: client/src/components/App.js [hunk 1]
```diff
@@ -1,10 +1,14 @@
 import React, { Component } from 'react';
 import 'materialize-css/dist/css/materialize.min.css';
+import 'materialize-css';
```
Comment (claireinez) [inline client/src/components/App.js]: Why do you need both these imports?

File: client/src/components/App.js [hunk 2]
```diff
@@ -24,6 +30,18 @@ class App extends Component {
     });
   }
 
+  handleDetails(id) {
+    console.log(id);
```
Comment (claireinez) [inline client/src/components/App.js]: Remove this

File: client/src/components/App.js [hunk 3]
```diff
@@ -24,6 +30,18 @@ class App extends Component {
...
+    axios
+      .get(`/api/selected/${id}`)
+      .then(property => {
+        this.setState({ details: property.data });
+      })
+      .catch(err => {
+        console.log(err);
```
Comment (claireinez) [inline client/src/components/App.js]: If you have extra time, you could always add in some proper error handling :)

File: client/src/components/MapContainer.js [hunk 4]
```diff
@@ -66,13 +64,11 @@ class MapContainer extends Component {
...
           mapElement={<div style={{ height: '100%' }} />}
           markers={this.state.markers}
-
-
           onMarkerClick={this.handleMarkerClick}
           onInfoClose={this.handleInfoClose}
           defaultCenter={this.props.defaultCenter}
           defaultZoom={this.props.defaultZoom}
-
+          onPopClick={this.props.handlePopupClick}
```
Comment (claireinez) [inline client/src/components/MapContainer.js]: Perhaps this should be called `onPopupClick` to match the handler function name?

File: client/src/components/PropertyDetails.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            src={props.details.imageUrl}
+            alt=""
+          />
+        </li>
+        <li className="collection-item">
+          <div className="">Price in NIS:{props.details.price}</div>
+        </li>
+        <li className="collection-item">
+          <div className="">
+            USD:{props.details.price && Math.round(props.details.price / 3.42)}
```
Comment (claireinez) [inline client/src/components/PropertyDetails.js]: The exchange rate is not a constant...

File: controllers/sendMail.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+require('env2')('./config.env');
+
+module.exports = (req, res) => {
+  const { email, message, name } = req.body;
+  const transporter = nodemailer.createTransport({
+    service: 'gmail',
+    secure: false,
+    port: 25,
+    auth: {
+      user: 'noreply.hbnb@gmail.com',
```
Comment (claireinez) [inline controllers/sendMail.js]: This should be an environment variable along with the password

**Review by claireinez — CHANGES_REQUESTED** [review]

File: client/src/components/App.js (hunk 1 above)
Comment (hoslack) [reply client/src/components/App.js]: I didn't notice that. Instead, I have imported the minimum CSS and JS files for fast loading. It doesn't work when I just import the entire package. 
`import 'materialize-css/dist/js/materialize.min.js';`

`import 'materialize-css/dist/css/materialize.min.css';`

File: client/src/components/App.js (hunk 3 above)
Comment (hoslack) [reply client/src/components/App.js]: I hope returning a div with information is a better way. I have changed.

File: client/src/components/App.js (hunk 2 above)
Comment (hoslack) [reply client/src/components/App.js]: Done.

File: client/src/components/MapContainer.js (hunk 4 above)
Comment (hoslack) [reply client/src/components/MapContainer.js]: Done.

File: client/src/components/PropertyDetails.js (hunk 5 above)
Comment (hoslack) [reply client/src/components/PropertyDetails.js]: I have changed. Now I am using an API to get the current and latest exchange rates in real time.

File: controllers/sendMail.js (hunk 6 above)
Comment (hoslack) [reply controllers/sendMail.js]: Done.

File: client/src/components/App.js [hunk 7]
```diff
@@ -42,10 +60,21 @@ class App extends Component {
                 <MapContainer
                   defaultCenter={this.state.defaultCenter}
                   defaultZoom={this.state.defaultZoom}
+                  handlePopupClick={this.handleDetails}
```
Comment (hoslack) [inline client/src/components/App.js:63]: @claireinez done

File: client/src/components/PropertyDetails.js [hunk 8]
```diff
@@ -1,41 +1,67 @@
...
+  componentDidMount() {
+    axios
+      .get(`http://apilayer.net/api/live?access_key=${key}&currencies=USD,ILS&format=1`)
+      .then(res => {
+        const exchangeRate = res.data.quotes.USDILS;
+        this.setState({ exchangeRate });
+      })
+      .catch(err => {
+        return <div>Could not get the exchange Rate</div>;
+      });
```
Comment (hoslack) [inline client/src/components/PropertyDetails.js]: The API here @claireinez

File: client/src/components/PropertyDetails.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
+import('env2')('./config.env');
+const key = process.env.Currency_API_KEY;
```
Comment (claireinez) [inline client/src/components/PropertyDetails.js]: I didn't spot this the first time around. Can you make this environment variable UPPER_SNAKE_CASE please?

File: client/src/components/PropertyDetails.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+      exchangeRate: 0,
+    };
+  }
+
+  componentDidMount() {
+    axios
+      .get(`http://apilayer.net/api/live?access_key=${key}&currencies=USD,ILS&format=1`)
+      .then(res => {
+        const exchangeRate = res.data.quotes.USDILS;
+        this.setState({ exchangeRate });
```
Comment (claireinez) [inline client/src/components/PropertyDetails.js:20]: Nice!

**Review by claireinez — COMMENTED** [review]
Can you please make your environment variables UPPER_SNAKE_CASE please? Sorry, I missed that the first time round. After that, I'm happy!

File: client/src/components/PropertyDetails.js (hunk 9 above)
Comment (hoslack) [reply client/src/components/PropertyDetails.js]: Done

**Review by claireinez — APPROVED** [review]

**Pull request merged**